src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpLogSerializer.cs

```diff
@@ -0,0 +1,282 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.Internal;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Serializer;
+
+internal class ProtobufOtlpLogSerializer
+{
+    private const int ReserveSizeForLength = 4;
+    private const int TraceIdSize = 16;
+    private const int SpanIdSize = 8;
+
+    private static readonly Stack<List<LogRecord>> LogsListPool = [];
+    private static readonly Dictionary<string, List<LogRecord>> ScopeLogsList = [];
+
+    internal static int WriteLogsData(byte[] buffer, int writePosition, SdkLimitOptions sdkLimitOptions, ExperimentalOptions experimentalOptions, Resources.Resource? resource, Batch<LogRecord> logRecordBatch)
```

Comment:
nit:

Pass in the batch by reference: `in Batch<LogRecord> logRecordBatch`

Can also be done in `ProtobufOtlpTraceSerializer` sorry didn't catch that before.